### PR TITLE
Refactor Dist Output

### PR DESCRIPTION
**Pull request by kristoferbaxter**
## Motivations
1. `syntax` npm proposal, allowing bundlers to selectively choose modern JS for an expected module output.
2. Reduce the number of builds.
3. Explicitly remove minification, compression, and ensure debugging code is retained for AMP specific builds.

## Solution
Updated `dist` structure:

Screen Shot 2019-05-03 at 10 19 44 AM

Importantly each sub dependency is nested, to allow for the node resolution algorithm to correctly grab relatively modern JS versus transpiled to ES5 code.

The matrix for features enabled per build:

|  | default | debug | amp |
| :--- | :---: | :---: | :---: |
| minified | X |  |  |
| transpiled mirrored output | X | X | | 
| debugging output included | | X | X |

## Todo
- [ ] Include debugging output reduction script for consumers of the `amp` version.

### CODE REVIEW

**Comment by kristoferbaxter** [comment]
Will need to be a minor release since we're still pre 1.0.0, but changes the output names and locations.

File: config/rollup.main-thread.js [hunk 1]
```diff
@@ -36,77 +35,61 @@ const ESModules = [
...
-        allowConsole: DEBUG_BUNDLE_VALUE,
+        allowConsole: true,
       }),
-      MINIFY_BUNDLE_VALUE ? compiler() : null,
-      MINIFY_BUNDLE_VALUE ? terser() : null,
-    ].filter(Boolean),
+    ],
   },
   {
     input: 'output/main-thread/index.safe.js',
```
Comment (choumx) [inline config/rollup.main-thread.js]: Should we rename the input files to `index.amp.js` etc. too?

File: demo/filtered-hydration/index.html [hunk 2]
```diff
@@ -23,20 +19,8 @@
...
-    import {upgradeElement} from '/dist/index.safe.mjs';
-    upgradeElement(document.getElementById('upgrade-me'), '/dist/worker.safe.mjs');
+    import {upgradeElement} from '/dist/amp/main.mjs';
+    upgradeElement(document.getElementById('upgrade-me'), '/dist/amp/worker/worker.mjs');
   </script>
-  <script nomodule async=false defer>
-    document.addEventListener('DOMContentLoaded', function() {
-      MainThread.upgradeElement(document.getElementById('upgrade-me'), '/dist/worker.safe.js');
-    }, false);
-  </script>
```
Comment (choumx) [inline demo/filtered-hydration/index.html]: Intentionally removing nomodule support here?

**Comment by kristoferbaxter** [comment]
> OOC, what version number are you going to release this as?

Thought was to release as `0.11.0` since minors can reflect significant changes pre `1.0.0`. What do you think?

File: config/rollup.main-thread.js (hunk 1 above)
Comment (kristoferbaxter) [reply config/rollup.main-thread.js]: Good point, I'll fix this up.

File: demo/filtered-hydration/index.html (hunk 2 above)
Comment (kristoferbaxter) [reply demo/filtered-hydration/index.html]: Yes, the `amp` builds do not generate a `nomodule` output. The expectation is AMP as a consumer will perform the work of transpiling this output.

Do you think this is a concern?

Comment (choumx) [reply demo/filtered-hydration/index.html]: Right-o.

**Comment by choumx** [comment]
0.11.0 SGTM.

**Pull request merged**